pkg/operator/apis/monitoring/v1/types.go

```diff
@@ -209,6 +209,15 @@ type Authorization struct {
 	Credentials *corev1.SecretKeySelector `json:"credentials,omitempty"`
 }
 
+// BasicAuth sets the `Authorization` header on every scrape request with the
+// configured username.
+//
+// Currently the password is not configurable and always empty.
+type BasicAuth struct {
+	// The username for authentication.
+	Username string `yaml:"username" json:"username"`
```

Comment:
issue: Users of CRD might be confused why there is no password. Given no better way to document things, would add big TODO why password is not yet here in commentary. WDYT? (: